coalahtml/tree.py

```diff
@@ -0,0 +1,83 @@
+from collections import OrderedDict
+from queue import Queue
+import os
+
+
+class Node:
+
+    def __init__(self, path):
+        self.child = {}
+        self.parent = None
+        self.path = path
+        self.abspath = ""
+
+
+class Tree:
+
+    def __init__(self, file_list):
+        """
+        Create a root and build the graph.
+        """
+        self.root = Node(Tree._findroot(file_list))
+        self.root.abspath = self.root.path
+        self.fl_split = [path.split(os.sep) for path in file_list]
+        self._build(self.fl_split)
+
+    def _findroot(self, file_list):
```

Comment:
If you want to stick with your solution, I think it's better to use `os.path.split` for splitting paths. Main reason is that it's what docs suggest https://docs.python.org/3/library/os.html#os.sep